zephyr/port/bootloader/bootloader_flash.c

```diff
@@ -333,53 +340,93 @@ static esp_err_t bootloader_flash_read_allow_decrypt(size_t src_addr, void *dest
 
 esp_err_t esp_rom_flash_read(size_t src_addr, void *dest, size_t size, bool allow_decrypt)
 {
-    if (src_addr & 3) {
-        ESP_EARLY_LOGE(TAG, "bootloader_flash_read src_addr 0x%x not 4-byte aligned", src_addr);
-        return ESP_FAIL;
-    }
-    if (size & 3) {
-        ESP_EARLY_LOGE(TAG, "bootloader_flash_read size 0x%x not 4-byte aligned", size);
-        return ESP_FAIL;
-    }
-    if ((intptr_t)dest & 3) {
-        ESP_EARLY_LOGE(TAG, "bootloader_flash_read dest 0x%x not 4-byte aligned", (intptr_t)dest);
-        return ESP_FAIL;
-    }
+    uint8_t *dest_ptr = (uint8_t *)dest;
+    size_t aligned_size, copy_size;
+    size_t remaining = size;
+    uint8_t temp_buf[FLASH_BLK_SZ];
 
-    if (allow_decrypt) {
-        return bootloader_flash_read_allow_decrypt(src_addr, dest, size);
-    } else {
-        return bootloader_flash_read_no_decrypt(src_addr, dest, size);
+    /* handle misaligned source address */
+    size_t src_offset = src_addr & 3;
+    size_t start_addr = src_addr - src_offset;
+
+    while (remaining > 0) {
+        copy_size = (remaining > FLASH_BLK_SZ) ? FLASH_BLK_SZ : remaining;
```

Comment:
This should use `sizeof`, just in case the `temp_buf`'s size is changed in the future.

```suggestion
        copy_size = (remaining > sizeof(temp_buf)) ? sizeof(temp_buf) : remaining;
```